src/ellipse.rs

```diff
@@ -282,9 +301,149 @@ impl Shape for Ellipse {
     }
 }
 
+/// Calculates circumference C of an ellipse with radii (x, y) as the infinite series
+///
+/// C = pi (x+y) * ∑ binom(1/2, n)^2 * h^n from n = 0 to inf
+/// with h = (x - y)^2 / (x + y)^2
+/// and binom(.,.) the binomial coefficient
+///
+/// as described by Kummer ("Über die Hypergeometrische Reihe", 1837) and rediscovered by
+/// Linderholm and Segal ("An Overlooked Series for the Elliptic Perimeter", 1995).
+///
+/// The series converges very quickly for ellipses with only moderate eccentricity (`h` not close
+/// to 1).
+///
+/// The series is truncated to the sixth power, meaning a lower bound on the true value is
+/// returned. Adding the value of [`kummer_elliptic_perimeter_range`] to the value returned by this
+/// function calculates an upper bound on the true value.
+//
+// Note: Kurbo's minimum supported Rust version doesn't support const floating point arithmetic
+// yet, but this function can be made const on stable Rust.
+#[inline]
+fn kummer_elliptic_perimeter(radii: Vec2) -> f64 {
+    let Vec2 { x, y } = radii;
+    let h = ((x - y) / (x + y)) * ((x - y) / (x + y));
+    let h2 = h * h;
+    let h3 = h2 * h;
+    let h4 = h3 * h;
+    let h5 = h4 * h;
+    let h6 = h5 * h;
+
+    let lower = 1.
+        + h / 4.
+        + h2 / 64.
+        + h3 / 256.
+        + 25. * h4 / 16384.
```

Comment:
These would be slightly more efficient if written as `h4 * (25. / 16384.)` etc.

Reply:
Good point. I've lifted `PI` into that as well, plus a similar change for `fn kummer_elliptic_perimeter_range`. Only the purely variable operations are now done at runtime: 4e5e03f4165c9ffb9ffce56b435efa40de72eeba